pkg/deploy/taskdir/definitions/def_0_3.go

```diff
@@ -1070,6 +1073,28 @@ func (d Definition_0_3) addParametersToUpdateTaskRequest(ctx context.Context, re
 	return nil
 }
 
+func (d Definition_0_3) addResourcesToUpdateTaskRequest(ctx context.Context, client api.IAPIClient, req *api.UpdateTaskRequest) error {
+	if len(d.Resources) == 0 {
+		return nil
+	}
+
+	resourcesBySlug, err := getResourcesBySlug(ctx, client)
+	if err != nil {
+		return errors.Wrap(err, "fetching resources")
+	}
+
+	for alias, slug := range d.Resources {
+		r, ok := resourcesBySlug[slug]
+		if ok {
+			req.Resources[alias] = r.ID
+		} else {
+			return errors.Errorf("unknown resource: %s", slug)
```

Comment:
Prefer `%q` since it handles empty strings more clearly

```suggestion
			return errors.Errorf("unknown resource: %q", slug)
```